source/common/router/config_impl.h

```diff
@@ -215,6 +229,54 @@ class RouteEntryImplBase : public RouteEntry, public Matchable, public RedirectE
     uint64_t default_;
   };
 
+  /**
+   * Route entry implementation for weighted clusters.
+   * RouteEntryImplBase holds one or more weighted cluster objects,
+   * where each object has a back pointer to the parent
+   * RouteEntryImplBase object. Almost all functions in this class
+   * forward calls back to the parent, with the exception of clusterName
+   * and routeEntry.
+   */
+  struct WeightedClusterEntry : public RouteEntry, public Route {
```

Comment:
nit: please use normal layout of organizing overrides under a comment of the class they override, with any non-override methods at the top.